sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureDefaultTokenCredentialAutoConfiguration.java

```diff
@@ -34,13 +38,18 @@ public AzureDefaultTokenCredentialAutoConfiguration(AzureGlobalProperties azureG
     @Bean(name = DEFAULT_TOKEN_CREDENTIAL_BEAN_NAME)
     @ConditionalOnMissingBean(name = DEFAULT_TOKEN_CREDENTIAL_BEAN_NAME)
     @Order
-    public TokenCredential azureTokenCredential(AzureCredentialBuilderFactory<DefaultAzureCredentialBuilder> factory) {
+    public TokenCredential springDefaultAzureCredential(DefaultAzureCredentialBuilderFactory factory) {
         return factory.build().build();
     }
 
     @Bean
-    public AzureCredentialBuilderFactory<DefaultAzureCredentialBuilder> defaultAzureCredentialBuilderFactory() {
-        return new AzureCredentialBuilderFactory<>(azureGlobalProperties, new DefaultAzureCredentialBuilder());
+    public DefaultAzureCredentialBuilderFactory defaultAzureCredentialBuilderFactory(
```

Comment:
no ConditionalOnMissingBean?
also the return type should be AbstractAzureCredentialBuilderFactory and method name ( which is the implicated bean name) should be _azureCredentialBuilderFactory_

Reply:
OK, this `AbstractAzureCredentialBuilderFactory ` type will be more friendly.